Controllers/HomeController.cs

```diff
@@ -3,21 +3,88 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using System.Text.RegularExpressions;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Newtonsoft.Json;
 
 namespace Zoltu.Bags.Client.Controllers
 {
 	public class HomeController : Controller
 	{
+		private static Regex productRegx = new Regex(@"product\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex tagRegx = new Regex(@"tags\/(.*?)\/*?$", RegexOptions.IgnoreCase);
+		private static Regex minPriceRegx = new Regex(@"minprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex maxPriceRegx = new Regex(@"maxprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex aboutRegx = new Regex(@"aboutus", RegexOptions.IgnoreCase);
+		private static HttpClient client = null;
+
+		public HomeController()
+		{
+			// Best practices: https://www.asp.net/web-api/overview/advanced/calling-a-web-api-from-a-net-client
+			client = new HttpClient();
+			client.BaseAddress = new Uri("https://bags-api.zoltu.com/api/");
+			client.DefaultRequestHeaders.Accept.Clear();
+			client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+		}
+
 		[Route("")]
 		[Route("app/{*path}")]
-		public IActionResult App(string path = null)
+		public async Task<IActionResult> App(String path = null)
 		{
-			ViewData["url"] = "https://bagcupid.com/";
-			ViewData["type"] = "website";
-			ViewData["title"] = "Bag Cupid";
-			ViewData["description"] = "What is your dream bag? Are you having trouble finding it? Let us help you!";
+			var url = "https://bagcupid.com/";
+			var type = "website";
+			var title = "Bag Cupid";
+			var description = "What is your dream bag? Are you having trouble finding it? Let us help you!";
+			var image = "https://bagcupid.com/img/logo/bagcupid.png";
+
+			if (!String.IsNullOrEmpty(path))
```

Comment:
In this case, the performance hit isn't worth the complexity increase in my opinion, so I would prefer the null coalesce.  In some cases, performance wins are worth complexity increases while in others the performance wins aren't worth the complexity increases.  Async is a place where the performance gains are worth complexity increase because the potential gains from having a fully async call stack can be massive.  For this case, the additional cost of doing all of the matching doesn't outweigh the complexity increase of the extra conditional.